src/lang/parser.mly

```diff
@@ -324,12 +324,13 @@ meth_ty:
          | _ -> raise (Parse_error ($loc, "Invalid type constructor")) }
 
 ty_source:
-  | VARLPAR RPAR                  { mk_source_ty ~pos:$loc $1 [] }
-  | VARLPAR ty_source_params RPAR { mk_source_ty ~pos:$loc $1 $2 }
+  | VARLPAR RPAR                  { mk_source_ty ~pos:$loc ~extensible:false $1 [] }
+  | VARLPAR ty_source_params RPAR { mk_source_ty ~pos:$loc ~extensible:(fst $2) $1 (snd $2) }
 
 ty_source_params:
-  | VAR GETS ty_content { [$1,$3] }
-  | VAR GETS ty_content COMMA ty_source_params { ($1,$3)::$5 }
+  | VAR GETS ty_content { false, [$1,$3] }
+  | DOTDOTDOT { true, [] }
```

Comment:
You want to allow:
```liquidsoap
source(audio=pcm, ..., video=canvas)
```
This seems odd? I was more thinking:
```ocaml
ty_source:
  | VARLPAR RPAR                  { mk_source_ty ~pos:$loc ~extensible:false $1 [] }
  | VARLPAR DOTDOTDOT RPAR        { mk_source_ty ~pos:$loc ~extensible:true $1 [] } 
  | VARLPAR ty_source_params RPAR { mk_source_ty ~pos:$loc ~extensible:false $1 $2 }
  | VARLPAR ty_source_params COMMA DOTDOTDOT RPAR { mk_source_ty ~pos:$loc ~extensible:true $1 $2 }
```

Reply:
This is actually what I first wrote but you get a parser conflict if you write it this way...

And once you go to my solution, there is no real reason of for forbidding `...` in the middle (if you do you get more parser cases...)

Reply:
Cool that works for me then.